src/compiler/types.ts

```diff
@@ -1063,6 +1064,16 @@ namespace ts {
             | SyntaxKind.NeverKeyword;
     }
 
+    export interface ImportTypeNode extends TypeNode {
+        kind: SyntaxKind.ImportTypeNode;
+        isTypeOf?: boolean;

```

Comment:
Are you curious why I didn't modify type queries to allow import type nodes somehow? (Perhaps because you dislike 1-token loohaheads or think this feels inelegant?) Well over [here](https://github.com/weswigham/TypeScript/commit/d991f49d3de4a9d9fed4c6796cc49339463fcd5d) is a drop from awhile ago when I tried implementing this that way before, and it was really not pretty. Specifically, most places expect a type query's `exprName` to be an _Expression_ (or specifically an entity name), and some kind of import type node (and it is definitely a type node when it's not inside a type query, making the problem more difficult, since it ends up as both a `TypeNode` and an `Expression` branded node) is not usually an expression, which breaks expectations _everywhere_. Wrapping all the new behavior up into one new type (node) kind is the neatest way to handle the new feature, without breaking all existing consumers. 😄